docs-website/topics/serialization-json-io-sources.md

```diff
@@ -0,0 +1,280 @@
+    implementation "com.squareup.okio:okio:%okioVersion%"
+}
+```
+
+</tab>
+<tab id="maven" title="Maven">
+
+```xml
+<!-- pom.xml -->
+<dependencies>
+    <dependency>
+        <groupId>org.jetbrains.kotlinx</groupId>
+        <artifactId>kotlinx-serialization-json-io</artifactId>
+        <version>%serializationVersion%</version>
+    </dependency>
+    <dependency>
+        <groupId>org.jetbrains.kotlinx</groupId>
+        <artifactId>kotlinx-serialization-json-okio</artifactId>
+        <version>%serializationVersion%</version>
+    </dependency>
+    <dependency>
+        <groupId>org.jetbrains.kotlinx</groupId>
+        <artifactId>kotlinx-io-core</artifactId>
+        <version>%kotlinxIoVersion%</version>
+    </dependency>
+    <dependency>
+        <groupId>com.squareup.okio</groupId>
+        <artifactId>okio</artifactId>
+        <version>%okioVersion%</version>
+    </dependency>
+</dependencies>
+```
+</tab>
+</tabs>
+
+The next sections cover examples using `kotlinx-io` types with these APIs.  
+You can use Okio types similarly with their corresponding `BufferedSource` and `BufferedSink`.
+
+
+### Serialize JSON to Sinks
+
+Use the `.encodeToSink()` function to serialize JSON to a `Sink`:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+// Imports declarations for kotlinx-io types and JSON I/O support
+import kotlinx.serialization.json.io.*
+import kotlinx.io.*
+import kotlinx.io.files.*
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+@OptIn(ExperimentalSerializationApi::class)
+fun main() {
+    val project = Project("kotlinx.serialization", 9000)
+
+    // Creates a Sink for the project.json file
+    val path = Path("project.json")
+    SystemFileSystem.sink(path).buffered().use { sink: Sink ->
+
+        // Serializes the Project instance directly into a Sink
+        Json.encodeToSink(project, sink)
+    }
+}
+```
+
+### Deserialize JSON from Sources
+
+To deserialize JSON from a `Source`, use the `.decodeFromSource()` function:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+// Imports declarations for kotlinx-io types and JSON I/O support
+import kotlinx.serialization.json.io.*
+import kotlinx.io.*
+import kotlinx.io.files.*
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+@OptIn(ExperimentalSerializationApi::class)
+fun main() {
+    // Opens a Source for the project.json file
+    val path = Path("project.json")
+    SystemFileSystem.source(path).buffered().use { source: Source ->
+
+        // Deserializes a Project instance directly from a Source
+        val project = Json.decodeFromSource<Project>(source)
+
+        println(project)
+    }
+}
+```
+
+If your input contains a large JSON array or multiple top-level JSON objects,
```

Comment:
"a large JSON array or multiple top-level JSON objects" is a more correct version